src/rust/ensogl/lib/shape-utils/src/compound_shape.rs

```diff
@@ -0,0 +1,57 @@
+//! This module contains functionality for handling multiple shapes that should act like one.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl_core::gui::component::ShapeView;
+
+// ======================
+// === Compound Shape ===
+// ======================
+
+// FIXME `_dummy` input is needed as the input section can not be omitted.
+ensogl_text::define_endpoints! {
+    Input {
+        _dummy ()
+    }
```

Comment:
AFAIK it can have no inputs:
```suggestion
ensogl_text::define_endpoints! {
    Input {}
```